client/src/main/java/io/confluent/kafka/schemaregistry/client/MockSchemaRegistryClient.java

```diff
@@ -83,14 +83,15 @@ private int getIdFromRegistry(String subject, Schema schema, boolean registerReq
       if (schemaId == null) {
         id = id >= 0 ? id : ids.incrementAndGet();
```

Comment:
I think it's slightly better to do the assignment earlier and start using it in this branch instead of `id`. Eg: 
```java
schemaId = id >= 0 ? id : ids.incrementAndGet();
schemaIdCache.put(schema, schemaId);
```
Also, it's a not great that we were overwriting an input argument.